compiler/rustc_lexer/src/unescape.rs

```diff
@@ -58,19 +58,27 @@ pub enum EscapeError {
     NonAsciiCharInByteString,
 }
 
+/// Wranings that can be emitted during string unescaping.
+#[derive(Debug, PartialEq, Eq)]
+pub enum EscapeWarning {
+    /// After a line ending with '\', the next line contains whitespace
+    /// characters that are not skipped.
+    UnskippedWhitespaceAfterEscapedNewline,
+}
+
```

Comment:
In principle I'm ok with the approach of nested `Result`s, but in this case it feels like a better way would be to make `UnskippedWhitespaceAfterEscapedNewline` a variant of `EscapeError`. I know it muddles the waters a little bit by mixing warnings and errors, but I feel it is worth it due to the flattening of the code in this PR.

Reply:
Makes sense, I've changed it to be a single type.  Should I also change the name to EscapeDiagnostic, or is that too vague?

Reply:
I feel like `EscapeError` is slightly misleading, but it'll make more sense than `EscapeDiagnostic`.